### PR TITLE
feat: add reject sampling, magic reject sampling and join-prob reject sampling.

### PR DESCRIPTION
**Pull request by TaochunChern**
1. Impl reject sampler & join-prob reject sampler
2. impl triton version for this two reject sampler
3. add accuary & perf test for reject sampler & join prob sampler

### CODE REVIEW

File: tests/accuracy/operators/test_sampling.py [hunk 1]
```diff
@@ -44,6 +47,71 @@ def test_topp_filter(logits, topk, topp, min_tokens_to_keep):
...
     )
 
+@pytest.mark.parametrize(
+    "target_logits, draft_tokens, draft_probs, spec_step",
+    [(torch.randn((15, 4, 155136), dtype=torch.float32), torch.randint(0, 155136, (15, 3)), torch.ones((15, 3), dtype=torch.float32), 3)],
+)
+@auto_switch_platform()
+@bypass_not_implemented
+def test_reject_sampler(target_logits, draft_tokens, draft_probs, spec_step):
+    torch.manual_seed(42)
```
[the remark this answers is not in the document]
Comment (wwens7) [reply tests/accuracy/operators/test_sampling.py]: 噢 你是手动对比的

**Review by wwens7 — DISMISSED** [review]

File: mojo_opset/backends/ttx/kernels/npu/sample.py [hunk 2]
```diff
@@ -236,6 +236,176 @@ def top_p_filter_impl(
...
+    tl.store(batch_output_accept_lens_ptr, accept_len)
+
+def reject_sampling_impl(
+    target_probs: torch.Tensor, # [batch, spec_step + 1, vocab_size]
+    draft_tokens: torch.Tensor, # [batch, spec_step]
+    draft_probs: torch.Tensor,  # [batch, spec_step]
+    random_seed,
+) -> tuple[torch.Tensor, torch.Tensor]:
+    device = target_probs.device
+    batch_size, _, vocab_size = target_probs.shape
```
Comment (zhangjihang-BD) [inline mojo_opset/backends/ttx/kernels/npu/sample.py]: batch_size一般有多大，如果超过48的话，建议改成persistent kernel

**Review by zhangjihang-BD — APPROVED** [review]
